gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -759,10 +759,18 @@ private final PreparedStatement prepMinion(
                                           qinfo = qt.computeQueryInfo(qic);
                                           // Only rerouting selects to lead node. Inserts will be handled separately.
                                           // The below should be connection specific.
-                                          if (routeQuery && qinfo != null && qinfo.isSelect() && !isPreparedStatement()) {
+                                          if (routeQuery && qinfo != null && qinfo.isDML() && !isPreparedStatement()) {
                                             if (SnappyActivation.isColumnTable((DMLQueryInfo)qinfo, false)) {
-                                              return getPreparedStatementForSnappy(true, statementContext, lcc,
-                                                  false, checkCancellation);
+																							if (qinfo.isSelect()) {
```

Comment:
Fix the formatting. There are few extra lines and it seems the text has moved to the right, atleast the git shows it that way.